wazuh/wazuh-dashboard-plugins
# Propagate table updates to the dashboard on endpoints summary

## Description
This pull request adds a mechanism to update the dashboard visualizations in case events occur in the Endpoints summary table.
For this, a new `useService` hook is added with the corresponding tests that allows the service to be called again according to a refresh parameter.
The corresponding `DonutCard` and `OutdatedAgentsCard` tests are also updated
 
## Issues Resolved
- #6454 

## Evidence

[Evidence.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/eb3f4005-aee7-40f8-a537-340960df9241)


## Test

> [!NOTE]  
> For this test it is necessary to have more than one agent (not necessarily connected) and more than one group

**Steps to test:**
- Go to Server management -> Endpoints Summary
- Check that the module renders correctly
- Adding one or more agents to a group through the actions in the table should cause new requests to be made, updating the dashboard visualizations.
- Removing the group of one or more agents through the actions in the table should cause new requests to be made, updating the dashboard views.

### Check List
